src/Core/Content/Category/SalesChannel/CategoryRoute.php

```diff
@@ -138,4 +139,42 @@ private function createCriteria(string $pageId, Request $request): Criteria
 
         return $criteria;
     }
+
+    /**
+     * @return array<string, array<string, mixed>>|null
+     */
+    private function buildMergedCmsSlotConfig(CategoryEntity $category, SalesChannelContext $context): ?array
+    {
+        $inheritanceChain = $context->getLanguageIdChain();
+        if (\count($inheritanceChain) <= 1) {
+            return $category->getTranslation('slotConfig');
+        }
+
+        /** @var non-empty-list<string> $languageMergeOrder */
+        $languageMergeOrder = \array_reverse(\array_unique($inheritanceChain));
+        $translatedSlotConfigs = $this->getTranslatedSlotConfigs($category, $languageMergeOrder);
+
+        return \array_merge(...$translatedSlotConfigs);
```

Comment:
is the array merge required? getTranslatedSlotConfigs returns just an array, or? 🤔

Reply:
Of course, that's the most important part. This function created the inheritance hierarchy of all translations. Otherwise it would be a translation collection. 

getTranslatedSlotConfigs returns an array of config-arrays, which will be merged to the actual usage.